From 33c717a0655957603b44c590025f40070fda878d Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:52 -0400
Subject: [block] trace all devices plug operation

Message-id: <1402670238-13916-125-git-send-email-snitzer@redhat.com>
Patchwork-id: 83829
O-Subject: [RHEL7.1 PATCH 124/150] block: trace all devices plug operation
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 7aef2e780b13973ea60aed8c556107dabde6a495
Author: Jianpeng Ma <majianpeng@gmail.com>
Date:   Wed Sep 11 13:21:07 2013 -0600

    block: trace all devices plug operation

    In func blk_queue_bio, if list of plug is empty,it will call
    blk_trace_plug.
    If process deal with a single device,it't ok.But if process deal with
    multi devices,it only trace the first device.
    Using request_count to judge, it can soleve this problem.

    In addition, i modify the comment.

    Signed-off-by: Jianpeng Ma <majianpeng@gmail.com>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-core.c b/block/blk-core.c
index 2788b10..1410958 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -1637,11 +1637,9 @@ get_rq:
  if (plug) {
   /*
    * If this is the first request added after a plug, fire
-   * of a plug trace. If others have been added before, check
-   * if we have multiple devices in this plug. If so, make a
-   * note to sort the list before dispatch.
+   * of a plug trace.
    */
-  if (list_empty(&plug->list))
+  if (!request_count)
    trace_block_plug(q);
   else {
    if (request_count >= BLK_MAX_REQUEST_COUNT) {
-- 
1.7.1